transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -167,10 +168,14 @@ internal class MessagingClientImpl(
     }
 
     private fun handleError(code: ErrorCode, message: String? = null) {
-        if (code == ErrorCode.SessionHasExpired || code == ErrorCode.SessionNotFound) {
-            currentState = State.Error(code, message)
-        } else if (code == ErrorCode.MessageTooLong) {
-            messageStore.onMessageError(code, message)
+        when (code) {
+            is ErrorCode.SessionHasExpired,
+            is ErrorCode.SessionNotFound ->
+                currentState = State.Error(code, message)
+            is ErrorCode.MessageTooLong,
+            is ErrorCode.RequestRateTooHigh ->
+                messageStore.onMessageError(code, message)
```

Comment:
Right now Transport missing a proper error handling of 400-s errors. That's because,  Shyrka does not have a concept of request id so that transport can know what exact request failed. I was thinking about having a message queue. Messages (and here I mean requests) are going to be sent after previous one got a response. But this is out of scope of this pr. 
see: https://inindca.atlassian.net/browse/GMDX-179